src/theory/strings/array_core_solver.cpp

```diff
@@ -142,7 +142,8 @@ void ArrayCoreSolver::checkUpdate(const std::vector<Node>& updateTerms)
       Node lem = nm->mkNode(EQUAL, left, right);
 
       std::vector<Node> exp;
-      d_im.addToExplanation(termProxy, n, exp);
+      // we don't have to add (termProxy = n) to the explanation, since this
+      // is always true and justified by definition.
```

Comment:
Is is possible to add an assertion that verifies that?

Reply:
The assertion would be to check that the equality is true in the equality engine. However, this does not hold after this PR since we may be doing lazy term registration, where `termProxy` is not yet introduced in a lemma.